cmd/gentypes/gentypes.go

```diff
@@ -387,6 +387,33 @@ func emitMessageMethods(sb *strings.Builder, typeName string) {
 	}
 }
 
+// emitRequestMethods emits methods for typeName that make it implement the
+// RequestMessage interface. These methods are only emitted for top-level types which
+// should implement that interface; other types are ignored.
+func emitRequestMethods(sb *strings.Builder, typeName string) {
+	if strings.HasSuffix(typeName, "Request") && typeName != "Request" {
+		fmt.Fprintf(sb, "func (r *%s) GetRequest() *Request {return &r.Request}\n", typeName)
+	}
+}
+
+// emitResponseMethods emits methods for typeName that make it implement the
+// ResponseMessage interface. These methods are only emitted for top-level types which
+// should implement that interface; other types are ignored.
+func emitResponseMethods(sb *strings.Builder, typeName string) {
+	if strings.HasSuffix(typeName, "Response") && typeName != "Response" {
+		fmt.Fprintf(sb, "func (r *%s) GetResponse() *Response {return &r.Response}\n", typeName)
+	}
+}
+
+// emitEventMethods emits methods for typeName that make it implement the
+// EventMessage interface. These methods are only emitted for top-level types which
+// should implement that interface; other types are ignored.
+func emitEventMethods(sb *strings.Builder, typeName string) {
+	if strings.HasSuffix(typeName, "Event") && typeName != "Event" {
+		fmt.Fprintf(sb, "func (e *%s) GetProtocolMessage() *ProtocolMessage {return &e.ProtocolMessage}\n", typeName)
```

Comment:
Why does the new code return Request for requests, Response for responses, but ProtocolMessage (not Event) for events?

Reply:
That is a very good question. It missed that the Event-Types do embed an Event-Struct and thought they only had a ProtocolMessage embedded. (Do not ask why I though this. I do not really know myself...).
But it would of course make more sense to return an Event here. I will fix this up.